docs/ops/deployment/docker.md

```diff
@@ -24,119 +24,490 @@ under the License.
+
+    ENV VAR_NAME value
+    ```
+
+    ```sh
+    docker build -t custom_flink_image .
+    # optional push to your docker image registry if you have it,
+    # e.g. to distribute the custom image to your cluster
+    docker push custom_flink_image
+    ```
+
+{% top %}
+
+## Flink with Docker Compose
+
+[Docker Compose](https://docs.docker.com/compose/) is a way to run a group of Docker containers locally.
+The next chapters show examples of configuration files to run Flink.
 
 ### Usage
 
+* Create the `yaml` files with the container configuration, check examples for:
+    * [Session cluster](#session-cluster-with-docker-compose)
+    * [Single Job](#single-job-cluster-with-docker-compose)
+
+    See also [the Flink docker image tags](#image-tags) and [how to customize the Flink docker image](#advanced-customization)
+    for usage in the configuration files.
+
 * Launch a cluster in the foreground
 
-        docker-compose up
+    ```sh
+    docker-compose up
+    ```
 
 * Launch a cluster in the background
 
-        docker-compose up -d
+    ```sh
+    docker-compose up -d
+    ```
 
-* Scale the cluster up or down to *N* TaskManagers
+* Scale the cluster up or down to *N TaskManagers*
 
-        docker-compose scale taskmanager=<N>
+    ```sh
+    docker-compose scale taskmanager=<N>
+    ```
 
-* Kill the cluster
+* Access the *Flink Master* container
 
-        docker-compose kill
+    ```sh
+    docker exec -it $(docker ps --filter name=jobmanager --format={{.ID}}) /bin/sh
+    ```
 
-When the cluster is running, you can visit the web UI at [http://localhost:8081](http://localhost:8081). 
-You can also use the web UI to submit a job to a session cluster.
+* Kill the cluster
 
-To submit a job to a session cluster via the command line, you must copy the JAR to the JobManager
-container and submit the job from there.
+    ```sh
+    docker-compose kill
+    ```
+
+* Access Web UI
+
+    When the cluster is running, you can visit the web UI at [http://localhost:8081](http://localhost:8081).
+    You can also use the web UI to submit a job to a *Session cluster*.
+
+* To submit a job to a *Session cluster* via the command line, you must copy the JAR to the *Flink Master* container and
+submit the job from there. For example:
+
+    ```sh
+    JOB_CLASS_NAME="com.job.ClassName"
+    MASTER_CONTAINER=$(docker ps --filter name=jobmanager --format={{.ID}})
+    docker cp path/to/jar "${MASTER_CONTAINER}":/job.jar
+    docker exec -t -i "${MASTER_CONTAINER}" flink run -d -c ${JOB_CLASS_NAME} /job.jar
+    ```
+
+### Session Cluster with Docker Compose
+
+```yaml
+version: "2.2"
+services:
+  jobmanager:
+    image: flink:{% if site.is_stable %}{{site.version}}-scala{{site.scala_version_suffix}}{% else %}latest{% endif %}
+    ports:
+      - "8081:8081"
+    command: jobmanager
+    environment:
+      - |
+        FLINK_PROPERTIES=
+        jobmanager.rpc.address: jobmanager
+        taskmanager.numberOfTaskSlots: 2
```

Comment:
curious; is this actually required for the JobManager?

Reply:
asaik, it only matters for active integrations atm, not for standalone, but it does not hurt and who knows future.